### PR TITLE
Set template and migrate apps to iOS 12 minimum

### LINKED ISSUE #136060: Targeting iOS 11 (Flutter default minimum) built with Xcode 15 crashes `EXC_BAD_ACCESS` when using the `Networking` framework
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Compile my add and try to run it on any iOS device with iOS less than 17

### Expected results

Should still run on all devices

### Actual results

The app will just close and not run. The version compiled on the previous Xcode runs on all iOS versions

### Code sample

I can not sare my code on a public forum but here you see when doing a flutter run it compiles and sends the app to the device. But will not run and does not give any error

```console
[        ] Executing commands in
'/tmp/11B6DEEF-BC32-4701-8E4A-2B4EC6C4D636/fruitstrap-lldb-prep-cmds-4781749bcc830a01bf3326c466d1d86cea0bc844'.
[        ] (lldb)     platform select remote-'ios' --sysroot '/Users/jacofourie/Library/Developer/Xcode/iOS DeviceSupport/12.5.7
(16H81)/Symbols'
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]    Sysroot: /Users/jacofourie/Library/Developer/Xcode/iOS DeviceSupport/12.5.7 (16H81)/Symbols
[        ]   SDK Path: "/Users/jacofourie/Library/Developer/Xcode/iOS DeviceSupport/12.5.7 (16H81)/Symbols"
[        ] (lldb)     target create "/Users/jacofourie/Desktop/FlutterProjects/smh/build/ios/iphoneos/Runner.app"
[+4832 ms] Current executable set to '/Users/jacofourie/Desktop/FlutterProjects/smh/build/ios/iphoneos/Runner.app' (arm64).
[        ] (lldb)     script
fruitstrap_device_app="/private/var/containers/Bundle/Application/F29C2CCD-BFBC-49C1-8363-8B854E50BB53/Runner.app"
[ +109 ms] (lldb)     script fruitstrap_connect_url="connect://127.0.0.1:64790"
[        ] (lldb)     script fruitstrap_output_path=""
[        ] (lldb)     script fruitstrap_error_path=""
[        ] (lldb)     target modules search-paths add /usr "/Users/jacofourie/Library/Developer/Xcode/iOS DeviceSupport/12.5.7
(16H81)/Symbols/usr" /System "/Users/jacofourie/Library/Developer/Xcode/iOS DeviceSupport/12.5.7 (16H81)/Symbols/System"
"/private/var/containers/Bundle/Application/F29C2CCD-BFBC-49C1-8363-8B854E50BB53"
"/Users/jacofourie/Desktop/FlutterProjects/smh/build/ios/iphoneos"
"/var/containers/Bundle/Application/F29C2CCD-BFBC-49C1-8363-8B854E50BB53"
"/Users/jacofourie/Desktop/FlutterProjects/smh/build/ios/iphoneos" /Developer "/Users/jacofourie/Library/Developer/Xcode/iOS
DeviceSupport/12.5.7 (16H81)/Symbols/Developer"
[  +26 ms] (lldb)     command script import
"/tmp/11B6DEEF-BC32-4701-8E4A-2B4EC6C4D636/fruitstrap_4781749bcc830a01bf3326c466d1d86cea0bc844.py"
[   +2 ms] (lldb)     command script add -f fruitstrap_4781749bcc830a01bf3326c466d1d86cea0bc844.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_4781749bcc830a01bf3326c466d1d86cea0bc844.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_4781749bcc830a01bf3326c466d1d86cea0bc844.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_4781749bcc830a01bf3326c466d1d86cea0bc844.safequit_command safequit
[        ] (lldb)     connect
[  +26 ms] (lldb)     run
[ +300 ms] success
[        ] (lldb)     safequit
[ +126 ms] Process 421 detached
[  +32 ms] Application launched on the device. Waiting for Dart VM Service url.
[ +721 ms] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
[+29288 ms] The Dart VM Service was not discovered after 30 seconds. This is taking much longer than expected...
[+85979 ms] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
```

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[+2892 ms] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
q 
quit
^C%                                                                                                                                       jacofourie@Jacos-Mac-Mini-M2 smh % flutter doctor -v
[✓] Flutter (Channel stable, 3.13.6, on macOS 14.0 23A344 darwin-arm64, locale en-ZA)
    • Flutter version 3.13.6 on channel stable at /Users/jacofourie/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ead455963c (9 days ago), 2023-09-26 18:28:17 -0700
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/jacofourie/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • Jaco’s iPhone 14 (mobile)    • 00008110-00110C9E1105401E                • ios            • iOS 17.0.3 21A360
    • Developmet iphone 6 (mobile) • 4781749bcc830a01bf3326c466d1d86cea0bc844 • ios            • iOS 12.5.7 16H81
    • macOS (desktop)              • macos                                    • darwin-arm64   • macOS 14.0 23A344 darwin-arm64
    • Chrome (web)                 • chrome                                   • web-javascript • Google Chrome 117.0.5938.149

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by jmagman**
1. Change templates to `IPHONEOS_DEPLOYMENT_TARGET`, `MinimumOSVersion`, and Podfile `platform :ios` to 12.0.
2. Add migrator for Podfile part to migrate `platform :ios, '11.0'` -> `platform :ios, '12.0'`
3. Compile with `-miphoneos-version-min=12.0`
4. Run the migrator on all example apps and integration tests.

See also https://github.com/flutter/flutter/pull/62902 and https://github.com/flutter/flutter/pull/85174 and https://github.com/flutter/flutter/pull/101963

Fixes https://github.com/flutter/flutter/issues/136060

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/integration_tests/ios_platform_view_tests/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -564,14 +564,12 @@
 				CURRENT_PROJECT_VERSION = 1;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				GENERATE_INFOPLIST_FILE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
```
Comment (jmagman) [inline dev/integration_tests/ios_platform_view_tests/ios/Runner.xcodeproj/project.pbxproj:567]: `PlatformViewUITests` target should not be overriding the project-level target version or device family.

File: examples/platform_view/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -311,7 +311,6 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				ENABLE_BITCODE = NO;
 				INFOPLIST_FILE = Runner/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
```
Comment (jmagman) [inline examples/platform_view/ios/Runner.xcodeproj/project.pbxproj:314]: `Runner` target should not be overriding the project-level target version.

File: packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart [hunk 3]
```diff
@@ -1172,7 +1172,7 @@ void main() {
       'CFBundleDisplayName': 'Awesome Gallery',
       // Will not use CFBundleName since CFBundleDisplayName is present.
       'CFBundleName': 'Awesome Gallery 2',
-      'MinimumOSVersion': '11.0',
+      'MinimumOSVersion': '17.0',
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart:1175]: This version can be anything for the sake of the test.

File: packages/flutter_tools/bin/podhelper.rb [hunk 4]
```diff
@@ -32,7 +32,7 @@ def flutter_additional_ios_build_settings(target)
   return unless target.platform_name == :ios
 
   # [target.deployment_target] is a [String] formatted as "8.0".
-  inherit_deployment_target = target.deployment_target[/\d+/].to_i < 11
+  inherit_deployment_target = target.deployment_target[/\d+/].to_i < 12
```
Comment (jmagman) [inline packages/flutter_tools/bin/podhelper.rb:35]: Everything before this is from the auto-migrate, this is the start of the real tool change.

**Review by stuartmorgan-g — APPROVED** [review]
LGTM

**Review by vashworth — APPROVED** [review]
LGTM

**Pull request merged**